client/state/selectors/can-upgrade-to-plan.js

```diff
@@ -28,6 +29,8 @@ export default function ( state, siteId, planKey ) {
 			: PLAN_FREE;
 	const plan = getCurrentPlan( state, siteId );
 
+	const planUpgradeIsLocked = isPurchaseManagementLocked( state, plan?.id );
```

Comment:
Here again we're treating a `plan` object as a purchase, does the plan really have an `is_locked` field? Also, in my testing `plan.id` is always `null` so the selector will always return `false`?

Reply:
I would expect `plan.id` to be empty either until all the page data had loaded, or if you didn't currently have a subscription to upgrade from (as this ID is strangely the subscription ID, mentioned above), e.g. for a site using the free plan, or for a Jetpack site with no paid Jetpack plans/subscriptions on it. Does this match your use case here or are you seeing something different?